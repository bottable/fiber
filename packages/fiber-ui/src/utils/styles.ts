import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
