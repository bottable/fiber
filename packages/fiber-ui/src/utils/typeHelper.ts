export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
