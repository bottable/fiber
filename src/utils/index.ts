export * from './typeHelper'
export * from './mergeProps'
