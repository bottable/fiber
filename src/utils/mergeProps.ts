import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
