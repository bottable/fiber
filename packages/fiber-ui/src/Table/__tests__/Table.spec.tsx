import { Table } from '..'

import React from 'react'
import { render } from 'test/utils'

describe('Table', () => {
  it('renders correctly', () => {
    const columns = [
      {
        title: 'Name',
        dataIndex: 'name',
        render: (text: string) => <a>{text}</a>
      },
      {
        title: 'Age',
        dataIndex: 'age'
      },
      {
        title: 'Address',
        dataIndex: 'address'
      }
    ]
    const data = [
      {
        key: '1',
        name: 'John Brown',
        age: 32,
        address: 'New York No. 1 Lake Park'
      },
      {
        key: '2',
        name: 'Jim Green',
        age: 42,
        address: 'London No. 1 Lake Park'
      },
      {
        key: '3',
        name: 'Joe Black',
        age: 32,
        address: 'Sidney No. 1 Lake Park'
      },
      {
        key: '4',
        name: 'Disabled User',
        age: 99,
        address: 'Sidney No. 1 Lake Park'
      }
    ]

    const table = render(<Table columns={columns} dataSource={data} />)

    expect(table).toMatchSnapshot()
  })
})
