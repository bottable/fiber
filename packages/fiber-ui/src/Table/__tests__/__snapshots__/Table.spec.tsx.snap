// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
<div
  className="sc-fzpkJw bVOhDm"
>
  <div
    className="sc-fznzOf hfODEb"
  >
    <table
      className="sc-fznMnq eblDLS"
    >
      <thead
        className="sc-fzoant cATlvS"
      >
        <tr>
          <th
            className="sc-fzqAbL dZCuAn"
          >
            Name
          </th>
          <th
            className="sc-fzqAbL dZCuAn"
          >
            Age
          </th>
          <th
            className="sc-fzqAbL dZCuAn"
          >
            Address
          </th>
        </tr>
      </thead>
      <tbody
        className="sc-fzomME jJsQZI"
        onChange={[Function]}
      >
        <tr
          className="sc-fzqMAW ggQcfJ"
        >
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            <a>
              John Brown
            </a>
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            32
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            New York No. 1 Lake Park
          </td>
        </tr>
        <tr
          className="sc-fzqMAW ggQcfJ"
        >
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            <a>
              Jim Green
            </a>
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            42
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            London No. 1 Lake Park
          </td>
        </tr>
        <tr
          className="sc-fzqMAW ggQcfJ"
        >
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            <a>
              Joe Black
            </a>
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            32
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            Sidney No. 1 Lake Park
          </td>
        </tr>
        <tr
          className="sc-fzqMAW ggQcfJ"
        >
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            <a>
              Disabled User
            </a>
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            99
          </td>
          <td
            className="sc-fzoydu gfzFEb"
            onClick={[Function]}
          >
            Sidney No. 1 Lake Park
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <ul
    className="sc-fzqBkg cAphmx sc-fznAgC ixcYbb"
  >
    <li
      className="sc-fzqPZZ jYHGWA"
    >
      <button
        className="sc-fzplWN sc-fzqBZW VzMMp sc-fzoxKX bRnqhF"
        disabled={true}
        onClick={[Function]}
        onMouseDown={[Function]}
        selected={false}
      >
        <span
          className="sc-fznyAO gfdBbl"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </span>
      </button>
    </li>
    <li
      className="sc-fzqPZZ jYHGWA"
      onClick={[Function]}
    >
      <button
        className="sc-fzplWN sc-fzqBZW bPWMnz sc-fzoxKX fnXnMJ"
        onClick={[Function]}
        onMouseDown={[Function]}
        selected={true}
      >
        <span>
          1
        </span>
      </button>
    </li>
    <li
      className="sc-fzqPZZ jYHGWA"
    >
      <button
        className="sc-fzplWN sc-fzqBZW VzMMp sc-fzoxKX bRnqhF"
        disabled={true}
        onClick={[Function]}
        onMouseDown={[Function]}
        selected={false}
      >
        <span
          className="sc-fznyAO gfdBbl"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
        </span>
      </button>
    </li>
  </ul>
</div>
`;
